src/lib/init.ts

```diff
@@ -19,7 +19,16 @@ export const initialize = async (): Promise<void> => {
       // Authed
       backupStatus = await getBackupStatus(program.session.fs)
 
-      const fullUsername = await program.components.storage.getItem(USERNAME_STORAGE_KEY) as string
+      let fullUsername = await program.components.storage.getItem(USERNAME_STORAGE_KEY) as string
+
+      // If the user is migrating from a version of webnative < 0.35.0, their username won't be in storage yet
+      // and won't contain a did, so we will need to manually append a DID and add it storage here
```

Comment:
Just a note, I think technically this isn't tied to the version of Webnative, but rather to the moment we introduced these new usernames: https://github.com/webnative-examples/webnative-app-template/pull/97/files#diff-a180510e798b8f833ebfdbe691c5ec4a1095076980d3e2388de29c849b2b8361R44

The username in the old storage (pre-webnative 0.35.0) is transferred over to the new storage location. Instead it is now found in the session db value (ie. `components.storage.KEYS.SESSION`): https://github.com/fission-codes/webnative/blob/main/src/index.ts#L715-L721

So if you want, you can get it from there too.

Reply:
good point, i'll clarify that in the comment